### PR TITLE
fix(subscriptions) return active subscriptions by default

### PR DESCRIPTION
**Pull request by ansmonjol**
## Context

We introduced a breaking change in the last release

`GET /subscription` was now returning all subscriptions no matter their `status`. But this endpoint is used to return only `active` ones.

## Description

This PR does fix that by 
1. returning only `active` subscription by default
2. keep allowing users to filter subscriptions by `status` and get the list is they want

### CODE REVIEW

File: app/queries/subscriptions_query.rb [hunk 1]
```diff
@@ -19,4 +20,17 @@ def with_external_customer(scope)
...
+    if filters.status.present? && valid_status?
+      scope.where(status: filters.status)
+    else
+      # Return active subscriptions by default
+      scope.where(status: [:active])
+    end
+  end
+
+  def valid_status?
+    filters.status.all? { |s| Subscription.statuses.key?(s) }
```
Comment (jdenquin) [inline app/queries/subscriptions_query.rb:34]: maybe you can add `filter.status.present?` here?

**Review by jdenquin — CHANGES_REQUESTED** [review]

File: app/queries/subscriptions_query.rb (hunk 1 above)
Comment (ansmonjol) [reply app/queries/subscriptions_query.rb:34]: I did it outside but can definitely move it in here, that would make sense

Comment (ansmonjol) [reply app/queries/subscriptions_query.rb:34]: done

File: app/queries/subscriptions_query.rb [hunk 2]
```diff
@@ -2,7 +2,8 @@
 
 class SubscriptionsQuery < BaseQuery
   def call
-    subscriptions = paginate(organization.subscriptions.active)
+    subscriptions = paginate(organization.subscriptions)
+    subscriptions = with_status(subscriptions)
```
[the remark this answers is not in the document]
Comment (ansmonjol) [reply app/queries/subscriptions_query.rb:6]: Love it.

But why should we push active and the end? That would mean that all requests by status will include active ones no?

That's not really what I would expect from this endpoint. If I ask for pending ones I would prefer to have only pending returned.

Will keep only the first part and not return it if that looks ok to you

```ruby
def filtered_statuses
  return [:active] unless valid_status?
  
  filter.status
end
```

[the remark this answers is not in the document]
Comment (jdenquin) [reply app/queries/subscriptions_query.rb]: oh yes if there is some status in params `:active` should not be always there, nice catch

[the remark this answers is not in the document]
Comment (ansmonjol) [reply app/queries/subscriptions_query.rb]: Ok! code is updated then

**Review by jdenquin — APPROVED** [review]

**Pull request merged**